File: parallel-rebuild.mjs
#!/usr/bin/env node

/**
 * Parallel Dashboard Reconstruction System
 * Spawns multiple LLM workers to reconstruct files concurrently
 * Each worker handles a specific component/file using different LLM providers
 */

import Anthropic from "@anthropic-ai/sdk";
import fs from "fs";
import path from "path";

const anthropic = new Anthropic({
  apiKey: process.env.ANTHROPIC_API_KEY,
});

const dashboardPath =
  "c:\\Users\\regan\\ID SYSTEM\\axiom-x\\constitutional-market-harmonics\\dashboard";

// Load documentation context
const readmeContent = fs.readFileSync(
  path.join(dashboardPath, "README.md"),
  "utf8"
);
const deploymentContent = fs.readFileSync(
  path.join(dashboardPath, "DEPLOYMENT_READY.md"),
  "utf8"
);
const liveApisContent = fs.readFileSync(
  path.join(dashboardPath, "LIVE_TRADING_APIS_SUMMARY.md"),
  "utf8"
);

// Worker configurations for parallel processing
const workers = [
  {
    id: "WORKER-SERVER",
    model: "claude-3-5-sonnet-20241022",
    file: "server.ts",
    path: "server.ts",
    description: "Express API Server with Socket.IO & Live APIs",
    context: `
## Express API Server (server.ts)

Core Requirements:
- Express 5.x framework
- Socket.IO for real-time updates (port 12345)
- 13 live data endpoints for trading
- SQLite database integration
- Finnhub API integration (primary)
- CORS enabled
- Error handling & logging
- Mock data fallback

Endpoints:
1. /api/live/quotes/:symbols - Real-time stock prices
2. /api/live/news/:symbol - Company news
3. /api/live/general-news - Market-wide news
4. /api/live/forex - Currency pair rates
5. /api/live/crypto - Cryptocurrency prices
6. /api/live/earnings/:symbol - Earnings calendar
7. /api/live/sentiment/:symbol - Analyst sentiment
8. /api/live/profile/:symbol - Company profile
9. /api/live/peers/:symbol - Competitor list
10. /api/live/insider/:symbol - Insider transactions
11. /api/live/ipo - IPO calendar
12. /api/live/economic-calendar - Economic events
13. /api/live/all-data - Bulk data endpoint

Legacy endpoints:
- /api/dashboard - Full dashboard data
- /api/portfolio - Portfolio positions
- /api/performance - Historical performance
- /api/trades - Trade history
- /api/chaos - Chaos attractor signals
- /api/constitutional - Constitutional scores

Real-time updates via Socket.IO every 5 seconds
    `,
  },
  {
    id: "WORKER-DASHBOARD",
    model: "claude-3-5-sonnet-20241022",
    file: "app/page.tsx",
    path: "app/page.tsx",
    description: "Main Dashboard Component - 25 Components, 8 Tabs",
    context: `
## Main Dashboard Component (app/page.tsx)

Architecture:
- Next.js 15 with React 18
- TypeScript strict mode
- Tailwind CSS dark theme
- Socket.IO real-time updates
- 25 components across 5 tiers
- 8 navigation tabs
- Error boundaries for resilience

Component Tiers:
TIER 1 (Always Visible - 5):
1. ResilientHeader - Portfolio value, ROI, constitutional score
2. ResilientPortfolioPanel - Holdings, cash, diversification
3. ResilientPerformancePanel - ROI charts, Sharpe ratio
4. ResilientActivityPanel - Recent activity, system health
5. ResilientNewsTicker - Live news updates

TIER 2 (Tab-Specific - 6):
6. ResilientChaosVisualizer - 3D chaos attractors
7. ResilientGlobalMarkets - 8+ exchange data
8. ResilientMarketSentiment - Sentiment analysis
9. ResilientConstitutionalRadar - 5 Yama alignment
10. ResilientNewsCarousel - News carousel
11. ResilientForexPanel - Currency trading

TIER 3 (Advanced Analysis - 7):
12. ResilientPanarchyCyclesGraph - Market cycles
13. ResilientAntenarrativeLens - Market narratives
14. ResilientFractalOptimizationPanel - Fractal analysis
15. ResilientAdvancedAnalyticsPanel - Advanced metrics
16. ResilientRiskAssessmentPanel - Risk analysis
17. ResilientChaosBottleneckOptimizer - Performance optimization
18. ResilientBalancesAndCashflow - Cash flow analysis

TIER 4 (Intelligence - 4):
19. ResilientChatInterface - Claude Sonnet chat overlay
20. ResilientConstitutionalNeuralNetwork - AI analysis
21. ConstitutionalRadar - Ethics scoring
22. ConstitutionalScorer - Alignment metrics

TIER 5 (Support - 3):
23. ResilientTradesFeed - Live trades
24. ErrorBoundary - Component protection
25. RootErrorBoundary - Global error handler

Tabs (8 total):
1. Overview - All 13 core components
2. Portfolio - Detailed positions
3. Performance - Historical analysis
4. Chaos - 3D visualizations
5. Global Markets - Multi-exchange data
6. News - News carousel & analysis
7. Chat - Full-screen Claude interface
8. Neural - Separate AI engine

Data Flow:
- WebSocket real-time updates
- API polling with fallback
- Mock data for development
- Local state management
- Responsive design (mobile-first)
    `,
  },
  {
    id: "WORKER-WEBSOCKET",
    model: "claude-3-5-sonnet-20241022",
    file: "hooks/useWebSocket.ts",
    path: "hooks/useWebSocket.ts",
    description: "WebSocket Hook - Message Batching & Parallel Routing",
    context: `
## WebSocket Integration Hook (hooks/useWebSocket.ts)

Advanced Real-time Architecture:
- Message batching (16-window for 94% API reduction)
- Parallel routing (4 concurrent streams)
- Backpressure management (50-100ms adaptive)
- Priority-based routing (high/normal/low)
- Connection resilience (auto-reconnect with exponential backoff)
- Real-time metrics tracking
- Sub-100ms latency target
- 1000+ messages/second throughput

Interface:
interface UseWebSocketOptions {
  url: string;
  reconnect?: boolean;
  reconnectInterval?: number;
  maxReconnectAttempts?: number;
  batchWindow?: number; // milliseconds
  priorityMode?: 'high' | 'normal' | 'low';
  backpressure?: {
    minDelay: number;
    maxDelay: number;
    adaptiveThreshold: number;
  };
}

Returns:
{
  data: any;
  connected: boolean;
  error: Error | null;
  send: (message: any, priority?: string) => void;
  metrics: {
    latency: number;
    throughput: number;
    backpressure: number;
    reconnects: number;
  };
}

Features:
1. Message Batching
   - Collect messages in 16ms windows
   - Send as single batch
   - Reduce API calls by ~94%

2. Parallel Routing
   - 4 concurrent processing streams
   - Route by message type
   - Distribute load evenly

3. Backpressure Management
   - Detect slow receivers
   - Adaptive throttling (50-100ms)
   - Prevent queue overflow

4. Priority Routing
   - High: Critical updates (immediate)
   - Normal: Regular updates (batched)
   - Low: Non-critical (throttled)

5. Connection Resilience
   - Auto-reconnect on disconnect
   - Exponential backoff
   - Message replay on reconnect

6. Performance Metrics
   - Track latency
   - Monitor throughput
   - Log backpressure events
   - Count reconnect attempts

Implementation:
- Socket.IO client integration
- Redux or Context for state
- React hooks best practices
- TypeScript strict typing
    `,
  },
  {
    id: "WORKER-LIB-LIVEAPIS",
    model: "claude-3-5-sonnet-20241022",
    file: "lib/liveDataApis.ts",
    path: "lib/liveDataApis.ts",
    description: "Live Data API Functions - 13 Endpoints",
    context: `
## Live Data APIs Library (lib/liveDataApis.ts)

13 Export Functions for Market Data:

1. fetchStockQuotes(symbols: string[])
   - Real-time prices for multiple symbols
   - Returns: { symbol, price, high, low, volume, bid, ask }

2. fetchCompanyNews(symbol: string, limit?: number)
   - Latest news for company
   - Returns: { headline, source, sentiment, timestamp }

3. fetchGeneralNews(limit?: number)
   - Market-wide news from 500+ sources
   - Returns: array of news items

4. fetchForexRates()
   - EUR/USD, GBP/USD, JPY/USD, AUD/USD, NZD/USD
   - Returns: { pair, bid, ask, timestamp }

5. fetchCryptoPrices(symbols?: string[])
   - Bitcoin, Ethereum, and other cryptos
   - Returns: { symbol, price, change24h, volume }

6. fetchEarningsCalendar(symbol?: string)
   - Earnings dates and estimates
   - Returns: { symbol, date, estimatedEPS, revenue }

7. fetchSentiment(symbol: string)
   - Analyst ratings and target prices
   - Returns: { rating, targetPrice, consensus }

8. fetchCompanyProfile(symbol: string)
   - Company info, sector, market cap
   - Returns: { name, website, logo, sector, marketCap }

9. fetchPeers(symbol: string)
   - Competitor companies
   - Returns: array of peer symbols

10. fetchInsiderTransactions(symbol: string)
    - CEO/Board trading activity
    - Returns: { name, role, action, shares, value }

11. fetchIPOCalendar(from?: string, to?: string)
    - Upcoming IPO dates
    - Returns: { company, date, priceRange, shares }

12. fetchEconomicCalendar()
    - Fed decisions, inflation, unemployment, GDP
    - Returns: { event, date, impact, forecast, actual }

13. fetchAllData(symbols: string[])
    - Bulk endpoint (all data at once)
    - Returns: { quotes, news, forex, crypto, earnings, sentiment }

Architecture:
- Primary: Finnhub API (40,000+ stocks, real-time)
- Secondary: Alpha Vantage (technical indicators)
- Caching layer (60 second TTL)
- Error handling with fallbacks
- Rate limit management
- Mock data for development

Configuration:
- FINNHUB_API_KEY from environment
- ALPHA_VANTAGE_API_KEY optional
- API URLs and endpoints
- Rate limit tracking
    `,
  },
  {
    id: "WORKER-COMPONENTS",
    model: "claude-3-5-sonnet-20241022",
    file: "components/Header.tsx",
    path: "components/Header.tsx",
    description: "Header Component - Portfolio Display & Status",
    context: `
## Header Component (components/Header.tsx)

Key Features:
- Portfolio value display (real-time)
- ROI percentage with trend indicator
- Constitutional score visualization
- Fractal love score display
- System health status indicator
- Last update timestamp
- Responsive mobile design
- Dark theme (slate palette)

Data Props:
{
  portfolioValue: number;
  roi: number;
  constitutionalScore: number;
  fractalLoveScore: number;
  systemHealth: 'healthy' | 'warning' | 'critical';
  lastUpdate: string;
}

Visual Elements:
1. Portfolio Value - Large, bold number with currency
2. ROI - Percentage with up/down indicator
3. Constitutional Score - 0-100 scale with color coding
4. Fractal Love Score - Visual meter
5. System Status - Indicator dot (green/yellow/red)
6. Time - "Updated X seconds ago"

Styling:
- Tailwind CSS dark theme
- Green for positive metrics
- Red for negative/alerts
- Yellow for warnings
- Animated transitions
- Responsive grid layout

Interactions:
- Click to drill down into details
- Hover for tooltips
- Auto-refresh via props
    `,
  },
  {
    id: "WORKER-COMPONENTS-PORTFOLIO",
    model: "claude-3-5-sonnet-20241022",
    file: "components/PortfolioPanel.tsx",
    path: "components/PortfolioPanel.tsx",
    description: "Portfolio Panel - Positions & Holdings",
    context: `
## Portfolio Panel Component (components/PortfolioPanel.tsx)

Features:
- Holdings table (symbol, shares, price, value, weight)
- Cash position display
- Total portfolio value
- Diversification metrics
- Position-level constitutional scoring
- Add/remove position buttons
- Sorting and filtering
- Mobile responsive table

Data Props:
{
  positions: Array<{
    symbol: string;
    shares: number;
    price: number;
    value: number;
    weight: number;
    constitutionalScore?: number;
  }>;
  cash: number;
  total: number;
}

Features:
1. Holdings Table
   - Symbol, Shares, Price, Value, Weight %
   - Sort by any column
   - Filter by sector/type
   - Constitutional score per position

2. Summary
   - Cash on hand
   - Total portfolio value
   - Number of positions
   - Diversification index

3. Actions
   - View position details
   - Edit position
   - Add new position
   - Remove position

Styling:
- Dark theme table
- Alternating row colors
- Hover effects
- Sort indicators
- Color-coded performance
    `,
  },
];

/**
 * Execute a single worker job
 */
async function executeWorker(worker) {
  console.log(`\n🚀 [${worker.id}] Starting: ${worker.description}`);

  const systemPrompt = `You are an expert TypeScript/React developer specializing in high-performance trading dashboards and financial applications.

You will reconstruct a production-ready file for Constitutional Market Harmonics Dashboard.

CRITICAL REQUIREMENTS:
1. Output ONLY valid TypeScript/TSX code
2. NO explanations, comments, or markdown
3. Include all necessary imports
4. Use TypeScript strict mode
5. Implement proper error handling
6. Follow React best practices
7. Use Tailwind CSS for styling
8. Make components resilient and reusable
9. Include proper typing with interfaces
10. Ensure code is production-ready

Do not include code blocks with backticks - just pure code.`;

  const userPrompt = `Reconstruct the file: ${worker.file}

${worker.context}

Output the complete, production-ready TypeScript code.`;

  try {
    const response = await anthropic.messages.create({
      model: worker.model,
      max_tokens: 8000,
      system: systemPrompt,
      messages: [
        {
          role: "user",
          content: userPrompt,
        },
      ],
    });

    const code =
      response.content[0].type === "text" ? response.content[0].text : "";

    // Clean up markdown if present
    let cleanCode = code;
    if (cleanCode.includes("```typescript")) {
      cleanCode = cleanCode
        .replace(/```typescript\n/g, "")
        .replace(/```tsx\n/g, "")
        .replace(/```\n/g, "")
        .replace(/```/g, "");
    }

    // Ensure directories exist
    const filePath = path.join(dashboardPath, worker.path);
    const dir = path.dirname(filePath);
    if (!fs.existsSync(dir)) {
      fs.mkdirSync(dir, { recursive: true });
    }

    // Write file
    fs.writeFileSync(filePath, cleanCode, "utf8");

    console.log(`✅ [${worker.id}] Completed: ${worker.file}`);
    return { id: worker.id, status: "success", file: worker.file };
  } catch (error) {
    console.error(`❌ [${worker.id}] Error:`, error.message);
    return { id: worker.id, status: "error", file: worker.file, error: error.message };
  }
}

/**
 * Main orchestration function
 */
async function main() {
  console.log("╔══════════════════════════════════════════════════════════╗");
  console.log(
    "║  PARALLEL DASHBOARD RECONSTRUCTION SYSTEM                 ║"
  );
  console.log(
    "║  Using Claude Sonnet 4.5 with Parallel Worker Architecture ║"
  );
  console.log("╚══════════════════════════════════════════════════════════╝");

  console.log(`\n📍 Target: ${dashboardPath}`);
  console.log(`🔧 Total Workers: ${workers.length}`);
  console.log(`⚡ Mode: Parallel Execution (Max 6 concurrent)\n`);

  // Execute all workers in parallel (batches of 3)
  const results = [];
  const batchSize = 3;

  for (let i = 0; i < workers.length; i += batchSize) {
    const batch = workers.slice(i, i + batchSize);
    console.log(`\n📊 Batch ${Math.floor(i / batchSize) + 1}/${Math.ceil(workers.length / batchSize)}`);

    const batchResults = await Promise.all(batch.map(executeWorker));
    results.push(...batchResults);

    // Small delay between batches to prevent rate limiting
    if (i + batchSize < workers.length) {
      console.log("⏳ Waiting before next batch...");
      await new Promise((resolve) => setTimeout(resolve, 2000));
    }
  }

  // Summary
  console.log("\n╔══════════════════════════════════════════════════════════╗");
  console.log("║  RECONSTRUCTION SUMMARY                                 ║");
  console.log("╚══════════════════════════════════════════════════════════╝\n");

  const successful = results.filter((r) => r.status === "success");
  const failed = results.filter((r) => r.status === "error");

  console.log(`✅ Successful: ${successful.length}/${workers.length}`);
  successful.forEach((r) => console.log(`   ✓ ${r.file}`));

  if (failed.length > 0) {
    console.log(`\n❌ Failed: ${failed.length}/${workers.length}`);
    failed.forEach((r) => console.log(`   ✗ ${r.file}: ${r.error}`));
  }

  console.log("\n🎉 Dashboard files reconstructed successfully!");
  console.log("\n📋 Next Steps:");
  console.log("   1. npm install");
  console.log("   2. npm run build");
  console.log("   3. npx tsx server.ts (Terminal 1)");
  console.log("   4. npm run dev (Terminal 2)");
  console.log("   5. Open http://localhost:3000");
}

main().catch(console.error);

File: parallel_rebuild.py
#!/usr/bin/env python3
"""
Parallel Dashboard Reconstruction System
Uses Claude Sonnet 4.5 to rebuild corrupted files in parallel
"""

import os
import json
import asyncio
import requests
from pathlib import Path
from concurrent.futures import ThreadPoolExecutor, as_completed

DASHBOARD_PATH = r"c:\Users\regan\ID SYSTEM\axiom-x\constitutional-market-harmonics\dashboard"
ANTHROPIC_API_KEY = os.environ.get("ANTHROPIC_API_KEY", "")

class LLMWorker:
    def __init__(self, worker_id, file_path, description, context):
        self.worker_id = worker_id
        self.file_path = file_path
        self.description = description
        self.context = context
        self.result = None
        
    def reconstruct(self):
        """Reconstruct file using Claude Sonnet 4.5"""
        print(f"\n🚀 [{self.worker_id}] Starting: {self.description}")
        
        system_prompt = """You are an expert TypeScript/React developer specializing in high-performance trading dashboards.

You will reconstruct a production-ready file for Constitutional Market Harmonics Dashboard.

CRITICAL:
1. Output ONLY valid TypeScript/TSX code
2. NO explanations or markdown backticks
3. Include all necessary imports
4. Use TypeScript strict mode
5. Implement error handling
6. Follow React best practices
7. Make components production-ready

Do not include ```typescript or ``` markers."""

        user_prompt = f"""Reconstruct the file: {self.file_path}

{self.context}

Output the complete, production-ready TypeScript code."""

        try:
            response = requests.post(
                "https://api.anthropic.com/v1/messages",
                headers={
                    "x-api-key": ANTHROPIC_API_KEY,
                    "anthropic-version": "2023-06-01",
                    "content-type": "application/json",
                },
                json={
                    "model": "claude-3-5-sonnet-20241022",
                    "max_tokens": 8000,
                    "system": system_prompt,
                    "messages": [{"role": "user", "content": user_prompt}],
                },
                timeout=60
            )
            
            if response.status_code == 200:
                data = response.json()
                code = data["content"][0]["text"]
                
                # Clean up markdown if present
                code = code.replace("```typescript\n", "").replace("```tsx\n", "").replace("```\n", "").replace("```", "")
                
                # Ensure directory exists
                full_path = Path(DASHBOARD_PATH) / self.file_path
                full_path.parent.mkdir(parents=True, exist_ok=True)
                
                # Write file
                with open(full_path, "w", encoding="utf-8") as f:
                    f.write(code)
                
                print(f"✅ [{self.worker_id}] Completed: {self.file_path}")
                self.result = {"status": "success", "file": self.file_path}
            else:
                error = response.json().get("error", {}).get("message", "Unknown error")
                print(f"❌ [{self.worker_id}] API Error: {error}")
                self.result = {"status": "error", "file": self.file_path, "error": error}
                
        except Exception as e:
            print(f"❌ [{self.worker_id}] Exception: {str(e)}")
            self.result = {"status": "error", "file": self.file_path, "error": str(e)}
        
        return self.result

# Worker definitions
workers = [
    LLMWorker(
        "WORKER-SERVER",
        "server.ts",
        "Express API Server with Socket.IO & 13 Live APIs",
        """
Express API Server (server.ts)

Core:
- Express 5.x on port 3001
- Socket.IO on port 12345
- SQLite database
- Finnhub API integration
- CORS enabled
- Error handling & logging
- Mock data fallback

13 Live Endpoints:
1. /api/live/quotes/:symbols - Stock prices
2. /api/live/news/:symbol - Company news
3. /api/live/general-news - Market news
4. /api/live/forex - Currency rates
5. /api/live/crypto - Crypto prices
6. /api/live/earnings/:symbol - Earnings calendar
7. /api/live/sentiment/:symbol - Analyst ratings
8. /api/live/profile/:symbol - Company info
9. /api/live/peers/:symbol - Competitors
10. /api/live/insider/:symbol - Insider trading
11. /api/live/ipo - IPO calendar
12. /api/live/economic-calendar - Economic events
13. /api/live/all-data - Bulk data

Legacy Endpoints:
- /api/dashboard, /api/portfolio, /api/trades, /api/performance, /api/chaos, /api/constitutional

Mock Data Generators:
- generateMockDashboard()
- generateMockPortfolio()
- generateMockTrades(limit)
- generateMockPerformance()
- generateConstitutionalScores()
- generateChaosSignals()

Real-time:
- Socket.IO broadcasts every 5 seconds
- Auto-reconnect with exponential backoff
- Event-based message handling
"""
    ),
    
    LLMWorker(
        "WORKER-DASHBOARD",
        "app/page.tsx",
        "Main Dashboard - 25 Components, 8 Tabs",
        """
Dashboard Component (app/page.tsx)

25 Components Across 5 Tiers:

TIER 1 (Always Visible - 5):
- Header: Portfolio, ROI, constitutional score
- PortfolioPanel: Holdings, cash, diversification
- PerformancePanel: Charts, Sharpe, ROI
- ActivityPanel: Trades, system health
- NewsTicker: Live news

TIER 2 (Tab-Specific - 6):
- ChaosVisualizer: Lorenz, Chen, Rössler 3D
- GlobalMarkets: 8+ exchanges
- MarketSentiment: Sentiment analysis
- ConstitutionalRadar: 5 Yama alignment
- NewsCarousel: News carousel
- ForexPanel: Currency trading

TIER 3 (Advanced - 7):
- PanarchyCyclesGraph: Market cycles
- AntenarrativeLens: Market narratives
- FractalOptimizationPanel: Fractal analysis
- AdvancedAnalyticsPanel: Advanced metrics
- RiskAssessmentPanel: Risk analysis
- ChaosBottleneckOptimizer: Performance
- BalancesAndCashflow: Cash flow

TIER 4 (Intelligence - 4):
- ChatInterface: Claude Sonnet overlay
- ConstitutionalNeuralNetwork: AI analysis
- ConstitutionalRadar: Ethics scoring
- ConstitutionalScorer: Alignment

TIER 5 (Support - 3):
- TradesFeed: Live trades
- ErrorBoundary: Component protection
- RootErrorBoundary: Global errors

8 Navigation Tabs:
1. Overview: All core components
2. Portfolio: Detailed positions
3. Performance: Historical analysis
4. Chaos: 3D visualizations
5. Global Markets: Multi-exchange
6. News: Carousel & analysis
7. Chat: Full-screen Claude
8. Neural: AI engine

Features:
- Real-time WebSocket updates
- API polling with fallback
- Mock data for development
- Responsive design
- Dark theme (slate)
- Error boundaries
- Performance optimization
"""
    ),
    
    LLMWorker(
        "WORKER-WEBSOCKET",
        "hooks/useWebSocket.ts",
        "WebSocket Hook - Batching & Parallel Routing",
        """
WebSocket Hook (hooks/useWebSocket.ts)

Advanced Features:
- Message batching: 16ms window, 94% API reduction
- Parallel routing: 4 concurrent streams
- Backpressure management: 50-100ms adaptive
- Priority routing: high/normal/low
- Auto-reconnect with exponential backoff
- Real-time metrics: latency, throughput, backpressure
- Sub-100ms latency target
- 1000+ msg/sec throughput

Hook Interface:
useWebSocket(url, options)
Returns: { data, connected, error, send, metrics }

Options:
{
  url: string;
  reconnect?: boolean;
  reconnectInterval?: number;
  maxReconnectAttempts?: number;
  batchWindow?: number;
  priorityMode?: 'high' | 'normal' | 'low';
  backpressure?: {
    minDelay: number;
    maxDelay: number;
    adaptiveThreshold: number;
  };
}

Message Batching:
- Collect messages in 16ms windows
- Send as single batch
- Reduce calls by ~94%

Parallel Routing:
- Route by message type
- 4 concurrent streams
- Distribute load evenly

Backpressure:
- Detect slow receivers
- Adaptive throttling
- Prevent queue overflow

Priority System:
- High: Immediate
- Normal: Batched
- Low: Throttled

Resilience:
- Auto-reconnect
- Message replay
- Exponential backoff

Metrics:
- Latency tracking
- Throughput monitoring
- Backpressure events
- Reconnect counts
"""
    ),
    
    LLMWorker(
        "WORKER-LIVEAPIS",
        "lib/liveDataApis.ts",
        "Live Data APIs - 13 Endpoints",
        """
Live Data APIs (lib/liveDataApis.ts)

13 Exported Functions:

1. fetchStockQuotes(symbols: string[])
   Returns: { symbol, price, high, low, volume, bid, ask }

2. fetchCompanyNews(symbol: string, limit?: number)
   Returns: { headline, source, sentiment, timestamp }

3. fetchGeneralNews(limit?: number)
   Returns: news array from 500+ sources

4. fetchForexRates()
   Returns: { EUR/USD, GBP/USD, JPY/USD, AUD/USD, NZD/USD }

5. fetchCryptoPrices(symbols?: string[])
   Returns: { symbol, price, change24h, volume }

6. fetchEarningsCalendar(symbol?: string)
   Returns: { symbol, date, estimatedEPS, revenue }

7. fetchSentiment(symbol: string)
   Returns: { rating, targetPrice, consensus }

8. fetchCompanyProfile(symbol: string)
   Returns: { name, website, logo, sector, marketCap }

9. fetchPeers(symbol: string)
   Returns: peer symbols array

10. fetchInsiderTransactions(symbol: string)
    Returns: { name, role, action, shares, value }

11. fetchIPOCalendar(from?: string, to?: string)
    Returns: { company, date, priceRange, shares }

12. fetchEconomicCalendar()
    Returns: { event, date, impact, forecast, actual }

13. fetchAllData(symbols: string[])
    Returns: bulk data { quotes, news, forex, crypto }

Architecture:
- Primary: Finnhub API (real-time)
- Secondary: Alpha Vantage
- Caching: 60 second TTL
- Error handling & fallbacks
- Rate limit management
- Mock data for development

Config:
- FINNHUB_API_KEY from env
- ALPHA_VANTAGE_API_KEY optional
"""
    ),
    
    LLMWorker(
        "WORKER-HEADER",
        "components/Header.tsx",
        "Header Component",
        """
Header Component (components/Header.tsx)

Display:
- Portfolio value (real-time)
- ROI % with trend
- Constitutional score
- Fractal love score
- System health
- Last update time

Props:
{
  portfolioValue: number;
  roi: number;
  constitutionalScore: number;
  fractalLoveScore: number;
  systemHealth: 'healthy' | 'warning' | 'critical';
  lastUpdate: string;
}

Visual:
- Large portfolio value
- ROI with up/down
- Score meters 0-100
- Status indicator
- Timestamp

Styling:
- Dark theme
- Green for positive
- Red for alerts
- Yellow for warnings
- Responsive grid

Interactions:
- Click for details
- Hover tooltips
- Auto-refresh
"""
    ),
]

def main():
    print("╔══════════════════════════════════════════════════════════╗")
    print("║  PARALLEL DASHBOARD RECONSTRUCTION - LLM WORKERS          ║")
    print("║  Using Claude Sonnet 4.5 with Parallel Architecture     ║")
    print("╚══════════════════════════════════════════════════════════╝")
    
    print(f"\n📍 Target: {DASHBOARD_PATH}")
    print(f"🔧 Total Workers: {len(workers)}")
    print(f"⚡ Mode: Parallel Execution (Max 3 concurrent)")
    print(f"🤖 Model: Claude 3.5 Sonnet")
    
    results = []
    
    # Execute workers in parallel batches
    with ThreadPoolExecutor(max_workers=3) as executor:
        futures = {executor.submit(worker.reconstruct): worker for worker in workers}
        
        for i, future in enumerate(as_completed(futures), 1):
            worker = futures[future]
            try:
                result = future.result()
                results.append(result)
            except Exception as e:
                print(f"❌ [{worker.worker_id}] Exception: {str(e)}")
                results.append({
                    "status": "error",
                    "file": worker.file_path,
                    "error": str(e)
                })
    
    # Summary
    print("\n╔══════════════════════════════════════════════════════════╗")
    print("║  RECONSTRUCTION SUMMARY                                 ║")
    print("╚══════════════════════════════════════════════════════════╝\n")
    
    successful = [r for r in results if r["status"] == "success"]
    failed = [r for r in results if r["status"] == "error"]
    
    print(f"✅ Successful: {len(successful)}/{len(workers)}")
    for r in successful:
        print(f"   ✓ {r['file']}")
    
    if failed:
        print(f"\n❌ Failed: {len(failed)}/{len(workers)}")
        for r in failed:
            print(f"   ✗ {r['file']}: {r.get('error', 'Unknown error')}")
    
    print("\n🎉 Reconstruction complete!")
    print("\n📋 Next Steps:")
    print("   1. npm install")
    print("   2. npm run build")
    print("   3. npx tsx server.ts (Terminal 1)")
    print("   4. npm run dev (Terminal 2)")
    print("   5. Open http://localhost:3000")

if __name__ == "__main__":
    main()
